packages/client/src/components/atoms/BookingsCountdown/AddToCalendar.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { useTranslation, ActionButton } from "@eisbuk/translations";
+import { useSelector } from "react-redux";
+import { DateTime } from "luxon";
+import { ICalendar } from "datebook";
+
+import makeStyles from "@mui/styles/makeStyles";
+
+import Button from "@mui/material/Button";
+
+import { LocalStore } from "@/types/store";
+import { __addToCalendarButtonId__ } from "@/__testData__/testIds";
+import { getAboutOrganization } from "@/store/selectors/app";
+
+interface Props {
+  /**
+   * Record of subscribed slots with subscribed slotIds as keys and subscribed duration as value.
+   * Doesn't need to be organized as we're checking for each value by key (no need for ordering and grouping).
+   */
+  bookedSlots?: LocalStore["firestore"]["data"]["bookedSlots"];
+}
+const AddToCalendar: React.FC<Props> = ({ bookedSlots = {} }) => {
+  const classes = useStyles();
+
+  const { displayName = "", location = "" } =
+    useSelector(getAboutOrganization) || {};
+  const { t } = useTranslation();
+
+  // is displayed when bookings are finalized aka deadline is null
+  const handleClick = () => {
+    let icalendar: ICalendar;
+
+    Object.values(bookedSlots).forEach((bookedSlot, i) => {
+      const startDate = getStartDate(bookedSlot.date, bookedSlot.interval);
+      const endDate = getEndDate(bookedSlot.date, bookedSlot.interval);
+
+      const bookedSlotEvent = {
+        title: `Booked Slot at ${displayName}`,
+        location: location,
+        start: startDate,
+        end: endDate,
+      };
+
+      /**
+       * ICalendar needs to be instantiated with an event
+       * which can't be an empty object so on the first iteration
+       * it's instantiated with the first bookedSlotEvent
+       */
+      if (i === 0) {
+        icalendar = new ICalendar(bookedSlotEvent).addProperty(
+          "UID",
+          `${bookedSlot.date}${bookedSlot.interval}`
+        );
+      } else {
+        /**
+         * To add another event to the calendar it must be of type ICalendar
+         */
+        icalendar
+          .addEvent(new ICalendar(bookedSlotEvent))
+          .addProperty("UID", `${bookedSlot.date}${bookedSlot.interval}`);
+      }
+      /**
+       * Download calendar on last iteration
+       */
+      if (i === Object.keys(bookedSlots).length - 1) {
+        icalendar.download("Booked_Slots.ics");
+      }
```

Comment:
Instead of counting `bookedSlots`' keys (and performing additional iteration, you can access the array you're already iterating over (`Object.values(bookedSlots)`), using a third `.forEach` param. i.e.
```typescript
 Object.values(bookedSlots).forEach((bookedSlot, i, originalArr) => {
      // ...rest of the code
      const arrLen = originalArr.length;
      // ...rest of the code
    });
```

Reply:
This way, you're not constructing a new array (`Object.keys(bookedSlots)`) just to count it.

Reply:
Also, is the last iteration check really necessary...you can just download the calendar after the `forEach` loop...`calendar` is still in scope.